src/arbitrary/_internals/ArrayArbitrary.ts

```diff
@@ -92,7 +82,7 @@ export class ArrayArbitrary<T> extends NextArbitrary<T[]> {
     // we accept a max of maxGeneratedLength consecutive failures. This circuit breaker may cause
     // generated to be smaller than the minimal accepted one.
     while (s.size() < N && numSkippedInRow < this.maxGeneratedLength) {
-      const current = this.safeGenerate(mrng, biasFactorItems, this.maxGeneratedLength);
```

Comment:
Actually `, this.maxGeneratedLength);` was a bug, fixed by the optimization of the code.